tests/libcudacxx_flag_test/libcudacxx_flag_test.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# Copyright (c) 2025, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+# Get the directory of this script
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+cd "${SCRIPT_DIR}"
+
+# Path to the RMM include directory (absolute path)
+RMM_INCLUDE_DIR="${SCRIPT_DIR}/../../include"
+
+echo "Testing compilation failure when LIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE is not defined"
+echo "Using RMM include directory: ${RMM_INCLUDE_DIR}"
+
+# Create a temporary file for compilation errors
+ERROR_FILE=$(mktemp)
+trap 'rm -f "${ERROR_FILE}"' EXIT
+
+# Try to compile the file without defining LIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE
+g++ -std=c++17 -I"${RMM_INCLUDE_DIR}" libcudacxx_flag_test.cpp -o libcudacxx_flag_test 2> "${ERROR_FILE}"
```

Comment:
```suggestion
set +e
g++ -std=c++17 -I"${RMM_INCLUDE_DIR}" libcudacxx_flag_test.cpp -o libcudacxx_flag_test 2> "${ERROR_FILE}"
set -e
```

I *think* this line is why this is currently failing in CI, like this:

```text
Testing compilation failure when LIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE is not defined
Using RMM include directory: /__w/rmm/rmm/tests/libcudacxx_flag_test/../../include
Error: Process completed with exit code 1.
```

([build link](https://github.com/rapidsai/rmm/actions/runs/13765081966/job/38489672536?pr=1852#step:9:6077))

You've set `set -e` at the top, which causes the script to exit with a non-0 exit code when any other command in the script does. This compile compile line is failing (as you expect!), but that `-e` is causing the script to exit before even entering this if-else block.